modules/Register/components/StudentRegistrationForm/StudentRegistrationForm.tsx

```diff
@@ -0,0 +1,316 @@
+import React from "react";
+
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Box, Button, Flex, PasswordInput, Select, Text, TextInput, Title } from "@mantine/core";
+import { Controller, useForm } from "react-hook-form";
+import { z } from "zod";
+
+import SubmitButton from "@/shared/components/SubmitButton";
+
+const StudentRegistrationSchema = z
+  .object({
+    firstName: z.string().nonempty("First Name is required"),
+    lastName: z.string().nonempty("Last Name is required"),
+    gender: z.enum(["Male", "Female"]),
+    address: z.string().nonempty("Address is required"),
+    phoneNumber: z.string().nonempty("Phone Number is required"),
+    email: z.string().email("Invalid email address"),
+    password: z.string().min(6, "Password must be at least 6 characters"),
+    confirmPassword: z.string().min(6, "Confirm Password is required"),
+    educationLevel: z.enum(["School", "College", "University"]),
+    medium: z.enum(["Bangla", "English"]).optional(),
+    class: z.string().optional(),
+    degree: z.enum(["Bachelors", "Masters"]).optional(),
+    degreeName: z.string().optional(),
+    semesterYear: z.string().optional(),
+  })
+  .refine((data) => data.password === data.confirmPassword, {
+    message: "Passwords don't match",
+    path: ["confirmPassword"],
+  });
+
+type TStudentRegistrationFormData = z.infer<typeof StudentRegistrationSchema>;
+
+const StudentRegistrationForm = () => {
+  const { control, handleSubmit, watch } = useForm<TStudentRegistrationFormData>({
+    resolver: zodResolver(StudentRegistrationSchema),
```

Comment:
```suggestion
    resolver: studentRegistrationSchemaResolver,
```


Put the validation schema inside a helpers file and import the validationSchemaResolver from that file.

Export would be something like this:
`export const studentRegistrationSchemaResolver = zodResolver(StudentRegistrationFormSchema)`